components/style/matching.rs

```diff
@@ -724,14 +721,13 @@ pub trait ElementMatchMethods : TElement {
                 Ok(shared_style) => {
                     // Yay, cache hit. Share the style.
                     let node = self.as_node();
-                    let style = &mut node.mutate_data().unwrap().style;
 
                     // TODO: add the display: none optimisation here too! Even
                     // better, factor it out/make it a bit more generic so Gecko
                     // can decide more easily if it knows that it's a child of
                     // replaced content, or similar stuff!
                     let damage =
-                        match node.existing_style_for_restyle_damage((*style).as_ref(), None) {
+                        match node.existing_style_for_restyle_damage(node.get_existing_style().as_ref(), None) {
```

Comment:
(to elaborate on my previous comment, you're, for example, creating a spurious refcount increment/decrement here which wasn't needed before)